test/cmocka/src/audio/mixer/mixer_test.c

```diff
@@ -56,18 +56,23 @@ struct comp_buffer *post_mixer_buf;
 int comp_register(struct comp_driver *drv)
 {
 	void *dst;
-
+	int err;
 	switch (drv->type) {
 	case SOF_COMP_MIXER:
 		dst = &mixer_drv_mock;
+		err = memcpy_s(dst, sizeof(mixer_drv_mock),
+			drv, sizeof(struct comp_driver));
 		break;
 
 	case SOF_COMP_MOCK:
 		dst = &drv_mock;
+		err = memcpy_s(dst, sizeof(drv_mock), drv,
+			sizeof(struct comp_driver));
 		break;
 	}
 
-	memcpy(dst, drv, sizeof(struct comp_driver));
+	if (err)
```

Comment:
```
error: 'err' may be used uninitialized in this function [-Werror=maybe-uninitialized]
```
as there's no `default:` label.

Can `drv->type` be neither SOF_COMP_MIXER nor SOF_COMP_MOCK here?

Reply:
I think there should be a default with a "can't happen" thing. An assert or return -EINVAL or something.

Reply:
Thanks a lot @paulstelian97 , I will submit that.